functional-tests/fn_bridge_deposit.py

```diff
@@ -42,33 +48,48 @@ def main(self, ctx: flexitest.RunContext):
         finalized_psbt = btcrpc.proxy.finalizepsbt(signed_psbt["psbt"])
         deposit_tx = finalized_psbt["hex"]
 
-        print(btcrpc.sendrawtransaction(deposit_tx))
+        original_num_deposits = len(seqrpc.strata_getCurrentDeposits())
+        print(f"Original deposit count: {original_num_deposits}")
+
+        reth = ctx.get_service("reth")
+        rethrpc = reth.create_rpc()
+
+        original_balance = int(rethrpc.eth_getBalance(f"0x{evm_addr}"), 16)
+        print(f"Balance before deposit: {original_balance}")
+
+        print("Deposit Tx:", btcrpc.sendrawtransaction(deposit_tx))
         # check if we are getting deposits
         wait_until(
-            lambda: len(seqrpc.strata_getCurrentDeposits()) > 0,
+            lambda: len(seqrpc.strata_getCurrentDeposits()) > original_num_deposits,
             error_with="seem not be getting deposits",
             timeout=SEQ_PUBLISH_BATCH_INTERVAL_SECS,
         )
 
-        reth = ctx.get_service("reth")
-        rethrpc = reth.create_rpc()
-        print(rethrpc.eth_blockNumber())
+        current_block_num = int(rethrpc.eth_blockNumber(), base=16)
+        print(f"Current reth block num: {current_block_num}")
 
         wait_until(
-            lambda: int(rethrpc.eth_getBalance(f"0x{evm_addr}"), 16) > 0,
-            error_with="zero eth balance",
+            lambda: int(rethrpc.eth_getBalance(f"0x{evm_addr}"), 16) > original_balance,
+            error_with="eth balance did not update",
             timeout=EVM_WAIT_TIME,
         )
 
-        block_num = rethrpc.eth_blockNumber()
+        deposit_amount = ROLLUP_PARAMS_FOR_DEPOSIT_TX["deposit_amount"] * SATS_TO_WEI
+
         balance = int(rethrpc.eth_getBalance(f"0x{evm_addr}"), 16)
-        assert (
-            balance == ROLLUP_PARAMS_FOR_DEPOSIT_TX["deposit_amount"] * SATS_TO_WEI
-        ), f"invalid deposit amount: {balance}"
+        print(f"Balance after deposit: {balance}")
+
+        net_balance = balance - original_balance
+        assert net_balance == deposit_amount, f"invalid deposit amount: {net_balance}"
 
-        wait_until(lambda: rethrpc.eth_blockNumber() > block_num, error_with="not building blocks")
+        wait_until(
+            lambda: int(rethrpc.eth_blockNumber(), base=16) > current_block_num,
```

Comment:
wait this returns a base16 number?
Is this exposed to users, should I document this in the docs?

Reply:
Ok, this is the underlying Reth RPC that we build upon.
Thanks! :)